### PR TITLE
[13005] Convert the PreferredDatePage to RTL testing

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
We want to rewrite our Enzyme based tests with tests in RTL that do not test implementation details.

## Testing done
Local and Unit

## Screenshots
n/a

## Acceptance criteria
- [ ] All tests are using RTL

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx [hunk 1]
```diff
@@ -1,133 +1,145 @@
...
-    const form = mount(
-      <PreferredDatePage
-        openFormPage={openFormPage}
-        history={history}
-        data={{}}
-      />,
+    expect(screen.baseElement).to.contain.text(
+      'What is the earliest date you’d like to be seen?',
     );
+    expect(screen.baseElement).to.contain.text(
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:40]: I think we'd also want to include an expect statement that verified that the date widget was on the page, probably by checking that there are two comboboxes and a textbox using `getByRole`.

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx [hunk 2]
```diff
@@ -1,133 +1,145 @@
...
+      store,
+    });
+
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /month/i }), [
+      '2',
+    ]);
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /day/i }), [
+      '2',
+    ]);
+    await userEvent.type(
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:93]: The docs suggest that we only need to use `await` if we're adding a delay in typing, so you could probably remove this.

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx [hunk 3]
```diff
@@ -1,133 +1,145 @@
...
-    );
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(<PreferredDatePage />, {
+      store,
+    });
 
-    expect(form.find('h1').text()).to.equal(pageTitle);
-    expect(document.title).contain(pageTitle);
-    form.unmount();
+    expect(screen.baseElement).to.contain.text(
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:141]: I think I'd put this expect in one of the other tests and remove this separate test.

**Review by jbalboni — COMMENTED** [review]
A couple of minor things, looks good overall

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx (hunk 2 above)
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:93]: @jbalboni I thought that too based on the docs but the IDE throws a warning about needing it 
image

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx (hunk 3 above)
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:141]: I agree. Changing now.

File: src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx (hunk 1 above)
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx:40]: Good call, adding checks now.

**Review by jbalboni — APPROVED** [review]

**Pull request merged**